applicationconfig/client/src/main/java/com/azure/applicationconfig/AzConfigClient.java

```diff
@@ -533,74 +533,16 @@ private AzConfigClient(AzConfigCredentials credentials, HttpPipeline pipeline) {
         // Closest to API goes first, closest to wire goes last.
-//        ArrayList<RequestPolicyFactory> factories = new ArrayList<>();
         List<HttpPipelinePolicy> policies = new ArrayList<HttpPipelinePolicy>();
 
         policies.add(new UserAgentPolicy(String.format("Azure-SDK-For-Java/%s (%s)", SDK_NAME, SDK_VERSION)));
         policies.add(new RequestIdPolicy());
         policies.add(new AzConfigCredentialsPolicy(credentials));
         policies.add(new RetryPolicy());
-//        policies.add(new RequestRetryPolicyFactory()); // todo - do we really need custom retry policy here?
-        policies.add(new HttpLoggingPolicy(pipelineOptions.httpLogDetailLevel()));
-
-        return new HttpPipeline(policies.toArray(new HttpPipelinePolicy[policies.size()]));
-    }
-
-    static class AzConfigCredentials {
-        private URL baseUri;
-        private String credential;
-        private byte[] secret;
+        policies.add(new HttpLoggingPolicy(logLevel));
 
-        URL baseUri() {
-            return baseUri;
-        }
-
-        String credential() {
-            return credential;
-        }
-
-        byte[] secret() {
-            return secret;
-        }
-
-        static AzConfigCredentials parseConnectionString(String connectionString) {
-            if (connectionString == null || connectionString.isEmpty()) {
-                throw new IllegalArgumentException(connectionString);
-            }
-
-            // Parse connection string
-            String[] args = connectionString.split(";");
-            if (args.length < 3) {
-                throw new IllegalArgumentException("invalid connection string segment count");
-            }
-
-            String endpointString = "endpoint=";
-            String idString = "id=";
-            String secretString = "secret=";
-
-            AzConfigCredentials credentials = new AzConfigCredentials();
-
-            for (String arg : args) {
-                String segment = arg.trim();
-                try {
-                    if (segment.toLowerCase().startsWith(endpointString)) {
-                        credentials.baseUri = new URL(segment.substring(segment.indexOf('=') + 1));
-                    } else if (segment.toLowerCase().startsWith(idString)) {
-                        credentials.credential = segment.substring(segment.indexOf('=') + 1);
-                    } else if (segment.toLowerCase().startsWith(secretString)) {
-                        String secretBase64 = segment.substring(segment.indexOf('=') + 1);
-                        credentials.secret = Base64.getDecoder().decode(secretBase64);
-                    }
-                } catch (MalformedURLException ex) {
-                    throw new IllegalArgumentException(ex);
-                }
-            }
-            return credentials;
-        }
+        return new HttpPipeline(policies.toArray(new HttpPipelinePolicy[0]));
```

Comment:
We should probably review HttpPipeline to make this API more convenient - accepting a List for example. CC @anuchandy

Reply:
Created issue: https://github.com/Azure/autorest-clientruntime-for-java/issues/575

Reply:
@JonathanGiles, @conniey just responded to the issue. Please take a look and share thoughts.